projects/step-core/src/lib/modules/basics/components/cron-selector/cron-selector.component.ts

```diff
@@ -1,22 +1,62 @@
-import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
-import { FormBuilder, FormControl } from '@angular/forms';
-import { Subject } from 'rxjs';
-import { DateTime } from 'luxon';
+import { Component, inject, Input, OnDestroy, TrackByFunction } from '@angular/core';
+import { ControlValueAccessor, NgControl } from '@angular/forms';
+import { noop } from 'rxjs';
+import { KeyValue } from '@angular/common';
+import { CRON_PRESETS } from '../../services/cron-presets.token';
+
+type OnChange = (value: string) => void;
+type OnTouch = () => void;
 
 @Component({
   selector: 'step-cron-selector',
   templateUrl: './cron-selector.component.html',
   styleUrls: ['./cron-selector.component.scss'],
-  exportAs: 'stepCronSelector',
 })
-export class CronSelectorComponent {
+export class CronSelectorComponent implements ControlValueAccessor, OnDestroy {
+  readonly _cronPresets = inject(CRON_PRESETS);
+  readonly trackByKeyValue: TrackByFunction<KeyValue<string, string>> = (index, item) => item.key;
+
   @Input() label?: string;
-  @Input() cronString?: string;
+  @Input() placeholder?: string;
+
+  private onChange: OnChange = noop;
```

Comment:
we could completely avoid using ```noop``` if we declared it like ```private onChange?: OnChange;``` and then invoked it like ```this.onChange?.('foo')```, also not quite sure what's the purpose of setting it back to ```noop``` in ```ngOnDestroy()```

Reply:
I mean it looks strange to me because they are after all component properties so it would make sense when the component gets destroyed and therefore not referenced for its properties to be garbage-collected